lib/ansible/modules/clustering/k8s.py

```diff
@@ -0,0 +1,529 @@
+    - Provide the YAML definition of an object when creating or updating.
+    type: dict
+  src:
+    description:
+    - Provide a path to a file containing the YAML definition of the object. Mutually
+      exclusive with I(resource_definition).
+    type: path
+  api_version:
+    description:
+    - Use to specify the API version when deleting an object, or when creating an object, such as namespace or
+      project, that only requires a I(name) attribute.
+    type: string
+  kind:
+    description:
+    - Use to specify the object kind when deleting an object, or when creating an object, such as namespace or
+      project, that only requires a I(name) attribute. For multi-word values, use snake_case. For example, 
+      DeploymentConfig would be passed as C(deployment_config).
+    type: string
+  name:
+    description:
+    - Use to specify an object name when deleting an object, or when creating an object, such as namespace or
+      project, that only requires a name attribute.
+    type: string
+  namespace: 
+    description:
+    - Use to specify the namespace or project when deleting an object, or when creating an object, such as namespace
+      or project, that only requires a I(name) attribute.
+    type: string
+  host:
+    description:
+    - Provide a URL for acessing the Kubernetes API.
+  api_key:
+    description:
+    - Token used to authenticate with the API.
+  kubeconfig:
+    description:
+    - Path to an existing Kubernetes config file. If not provided, and no other connection
+      options are provided, the openshift client will attempt to load the default
+      configuration file from I(~/.kube/config.json).
+    type: path
+  context:
+    description:
+    - The name of a context found in the Kubernetes config file.
+  username:
+    description:
+    - Provide a username for connecting to the API.
+  password:
+    description:
+    - Provide a password for connecting to the API. Use in conjunction with I(username).
+  cert_file:
+    description:
+    - Path to a certificate used to authenticate with the API.
+    type: path
+  key_file:
+    description:
+    - Path to a key file used to authenticate with the API.
+    type: path
+  ssl_ca_cert:
+    description:
+    - Path to a CA certificate used to authenticate with the API.
+    type: path
+  verify_ssl:
+    description:
+    - Whether or not to verify the API server's SSL certificates.
+    type: bool
+
+requirements:
+    - "python >= 2.7"
+    - "openshift >= 0.3"
+    - "PyYAML >= 3.11"
+'''
+
+EXAMPLES = '''
+- name: Create a k8s namespace
+  k8s:
+    name: testing
+    api_version: v1
+    kind: namespace
+    state: present
+
+- name: Create an OpenShift project
+  k8s:
+    name: testing
+    api_version: v1
+    kind: project
+    state: present
+
+- name: Create a Service object from an inline definition
+  k8s:
+    state: present
+    resource_definition:
+      apiVersion: v1
+      kind: Service
+      metadata:
+        name: web
```

Comment:
Curious what the behavior would be if the name/namespace/apiversion/kind were provided at the top level and here and did not match.

Reply:
Great questions. The individual values of `kind`, `name`, `namespace`, `api_version` will be overwritten by those found in a `resource_definition` or a config read in from `src`.

I updated the doc strings for both the `resource_definition` and `src` options. 

Thanks!